nhais-adaptor/edifact/incoming/parser/creators.py

```diff
@@ -0,0 +1,60 @@
+from edifact.incoming.models.interchange import InterchangeHeader
+from edifact.incoming.models.message import MessageSegmentRegistrationDetails, MessageSegmentBeginningDetails, \
+    MessageSegmentPatientDetails
+
+
+def create_interchange_header(interchange_header_dict):
+    """
+    Creates an incoming interchange header from the interchange header dictionary
+    Since the interchange header details are all in the one line we extract the details
+    from here
+    :param interchange_header_dict: Will just be a list of 1
+    containing a key value tuple of the interchange header details
+    :return: InterchangeHeader: The incoming representation of the edifact interchange header
+    """
+    header_segment = interchange_header_dict[0]
+    header_segment_value = header_segment[1]
+    header_segment_values = header_segment_value.split("+")
+    sender = header_segment_values[1]
+    recipient = header_segment_values[2]
+    date_time = header_segment_values[3]
+    return InterchangeHeader(sender, recipient, date_time)
+
+
+def create_message_segment_beginning(message_beginning_dict):
+    """
+    Creates an incoming message beginning from the message beginning dictionary
+    :param message_beginning_dict: The dictionary will contain a list of lines relevant to the
+    message beginning section BGM
+    :return: MessageSegmentBeginningDetails: The incoming representation of the edifact message beginning details
+    """
+    reference_segment = [value for key, value in message_beginning_dict if key == "RFF"][0]
+    reference_values = reference_segment.split(":")
+    reference_number = reference_values[1]
+    return MessageSegmentBeginningDetails(reference_number)
+
+
+def create_message_segment_registration(message_registration_dict):
+    """
+    Creates an incoming message registration from the message registration dictionary
+    :param message_registration_dict: The dictionary will contain a list of lines relevant to the
+    message registration section S01
+    :return: MessageSegmentRegistrationDetails: The incoming representation of the edifact message registration details
+    """
+    transaction_segment = [value for key, value in message_registration_dict if key == "RFF"][0]

```

Comment:
Is this not just effectively `message_registration_dict.get('RFF')[0]`? Or am I misunderstanding some logic

Reply:
Yes unfortunately it is a list of tuples since it is possible for it to contain duplicate keys.

Reply:
I'm going to extract this out to a little function though. Since it's doing a fair bit in the one line and it's reused in a few places in this file too

Reply:
Yeah that is a little unfortunate, I just had a quick look and python has some support for type 'suggestions', which might be helpful here just to clearly specify that the dict passed in isn't a standard python dictionary

Reply:
That's interesting. 
I've added a new Type `EdifactDict = NewType("EdifactDict", List[Tuple[str, str]])`
Which I have then used to specify in the args and return type this is the Type we are expecting.
I didn't know this could be done in Python and I like that you can be explicit about the Typing